Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -114,22 +114,31 @@ extension BrowserViewController: TopToolbarDelegate {
         if !handleIPFSSchemeURL(url, visitType: .unknown) {
           tabManager.selectedTab?.reload()
         }
-      } else if !isPrivateMode, url.domainURL.schemelessAbsoluteDisplayString.endsWithSupportedSNSExtension, let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode) {
-        Task { @MainActor in
-          let currentStatus = await rpcService.snsResolveMethod()
-          switch currentStatus {
-          case .ask:
-            // show name service interstitial page
-            showSNSDomainInterstitialPage(originalURL: url, visitType: .unknown)
-          case .enabled:
-            if let resolvedURL = await resolveSNSHost(url.schemelessAbsoluteDisplayString, rpcService: rpcService) {
+      } else if !isPrivateMode,
+                let url = topToolbar.currentURL,
+                DecentralizedDNSHelper.isSupported(domain: url.domainURL.schemelessAbsoluteDisplayString),
+                let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode),
+                let decentralizedDNSHelper = DecentralizedDNSHelper(
+                  rpcService: rpcService,
+                  ipfsApi: braveCore.ipfsAPI,
+                  isPrivateMode: isPrivateMode
+                ) {
+        topToolbarDidPressReloadTask?.cancel()
+        topToolbarDidPressReloadTask = Task { @MainActor in
+          topToolbar.locationView.loading = true
+          let result = await decentralizedDNSHelper.lookup(domain: url.schemelessAbsoluteDisplayString)
+          topToolbar.locationView.loading = tabManager.selectedTab?.loading ?? false
```

Comment:
Is this part checking which operation will finish first and set the loading according to it ?

Reply:
Yeah this sets loading to true, then performs a network request and then sets the loading state to the webview's loading state. At this stage, the webview loading state _should_ still be `false` because we haven't provided a url request to the webview to load yet. Previously, we only update the loading state based on the webview's loading state, so it felt safer to use that webview loading state logic to determine the loading state here and avoid any potential race conditions.